src/test/java/net/glowstone/entity/GlowPlayerTest.java

```diff
@@ -310,12 +312,19 @@ public void testGiveExpLevels() {
     @Test
     public void testGiveExp() {
         entity.giveExp(20);
+        boolean expEvent = true;
         verify(eventFactory).callEvent(argThat(input -> {
-            PlayerLevelChangeEvent event = (PlayerLevelChangeEvent) input;
```

Comment:
The test should still specify which class of message is fired, and it should have cases that create both. If this specific case creates both types of message in an unspecified order, then verify it twice, add the argument to a list in the matcher, and then run the relevant asserts against the list.

Reply:
Done